src/webots/gui/WbDragTransformEvent.cpp

```diff
@@ -215,8 +215,8 @@ void WbDragTranslateAlongAxisEvent::apply(const QPoint &currentMousePosition) {
   mViewDistanceUnscaling = mViewpoint->viewDistanceUnscaling(mSelectedTransform->position());
 
   WbMatrix4 matrix(mSelectedTransform->matrix());
-  const WbVector3 absoluteScale = matrix.scale();
-  matrix.scale(1.0f / absoluteScale.x(), 1.0f / absoluteScale.y(), 1.0f / absoluteScale.z());
+  WbVector3 scale = mSelectedTransform->scale();
```

Comment:
Can't we use const here?
```suggestion
  const WbVector3 scale = mSelectedTransform->scale();
```
Or even better:
```suggestion
  const WbVector3 &scale = mSelectedTransform->scale();
```